src/sagas/wallet.js

```diff
@@ -122,7 +122,17 @@ export function* isWalletServiceEnabled() {
     yield call(() => AsyncStorage.removeItem(IGNORE_WS_TOGGLE_FLAG));
   }
 
-  const walletServiceEnabled = yield call(checkForFeatureFlag, WALLET_SERVICE_FEATURE_TOGGLE);
+  let walletServiceEnabled = yield call(checkForFeatureFlag, WALLET_SERVICE_FEATURE_TOGGLE);
+
+  // At this point the networkSettings already have been set by startWallet.
+  const networkSettings = yield select(getNetworkSettings);
+  if (walletServiceEnabled && isEmpty(networkSettings.walletServiceUrl)) {
```

Comment:
I think the user experience for this is bad

I think we should add a checkbox in the network settings screen, "Use wallet-service if available?", validate the wallet-service URL if it is checked while saving

Reply:
But since this is a bug, I think we can do this in a future PR, if you both agree with my suggestion

@pedroferreira1 @alexruzenhack

Reply:
Here the KTLO: https://github.com/HathorNetwork/hathor-wallet-mobile/issues/471